### PR TITLE
[cling] Force flush cout after execution

### PR DESCRIPTION
**Pull request by vgvassilev**
The user might use utilities which print on cout and expects the output
to be shown immediately.

This patch automatically flushes std::cout after each execution of a wrapper.

### CODE REVIEW

File: interpreter/cling/lib/Interpreter/Interpreter.cpp [hunk 1]
```diff
@@ -1084,6 +1085,10 @@ namespace cling {
     utils::Analyze::maybeMangleDeclName(FD, mangledNameIfNeeded);
     IncrementalExecutor::ExecutionResult ExeRes =
        m_Executor->executeWrapper(mangledNameIfNeeded, res);
+
+    // Force-flush as we might be printing on screen with printf.
+    std::cout.flush();
```
Comment (Axel-Naumann) [inline interpreter/cling/lib/Interpreter/Interpreter.cpp]: This happens to also flush by `stdout` on most implementations, but I think it's not required? Should we add `fflush(stdout)`?

File: interpreter/cling/test/Prompt/Regression.C [hunk 2]
```diff
@@ -79,3 +79,13 @@ class MyClass;
 extern MyClass* my;
 class MyClass {public: MyClass* getMyClass() {return 0;}} cl;
 MyClass* my = cl.getMyClass();
+
+//
+printf("Auto flush printf\n");
```
Comment (Axel-Naumann) [inline interpreter/cling/test/Prompt/Regression.C]: See above - not sure the current code should be expected to flush this.

**Comment by Axel-Naumann** [comment]
This is strictly an improvement, thanks @vgvassilev ! The ref files need to be touched, but for the better...

**Comment by vgvassilev** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by vgvassilev** [comment]
@phsft-bot build on ROOT-performance-centos7-multicore/default with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Pull request merged**